addons/godot-next/references/bit_flag.gd

```diff
@@ -0,0 +1,92 @@
+# BitFlag
+# author: xdgamestudios
+# license: MIT
+# description: A class that allows to convert enums into bit flag enums and
+#              provides abstraction API to work with them.
+# todo: Implement additional features
+# usage:
+# - Initial setup:
+#   enum my_enum { a, b, c }
+#
+# - Creating:
+#     bf = BitFlag.new(my_enum) # Creates a new BitFlag given an enum.
+# - Flag access:
+#     bf.<name> = true/false # Allows to set the current bit
+# - Set/Put flag:
+#     bf.put(bf.a) # Enables the flag 'a'. Another BitFlag can be provided.
+# - Clear flag:
+#     bf.clear(bf.b) # Disables the flag 'b'. Another BitFlag can be provided.
+# - Toggle flag:
+#     bf.toggle(bf.b) # Inverts the flag 'b'. Another BitFlag can be provided.
+# - Check flag:
+#     bf.check(bf.b) # Checks flag 'b'. Another BitFlag can be provided.
+# - Get flag names:
+#     bf.get_active() # Returns an array of active names.
+#     bf.get_all() # Returns an array of all flags names.
+tool
+extends Reference
+class_name BitFlag
+
+##### PROPERTIES #####
+
+var _enum: Dictionary = {}
+var _flags: int = 0
+
+##### NOTIFICATIONS #####
+
+func _init(p_enum: Dictionary):
+	for a_key in p_enum:
+		_enum[a_key] = 1 << p_enum[a_key]
+
+func _get(p_property: String) -> int:
+	if _enum.has(p_property):
+		return _enum[p_property]
+	return 0
+
+func _set(p_property: String, p_value: bool) -> bool:
+	if _enum.has(p_property):
+		if p_value:
+			_flags |= _enum[p_property]
+		else:
+			_flags &= _enum[p_property]
+		return true
+	return false
+
+##### PUBLIC METHODS #####
+
+func put(p_value) -> int:
+	var flags: int = p_value._flags if get_script() == p_value.get_script() else p_value
```

Comment:
Add for potential errors, since you are doing `p_value.get_script()`:

    if not p_value:
        return

Reply:
For assign, clear, toggle, and check too